x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -2977,9 +2981,20 @@ export class RulesClient {
 
     this.ruleTypeRegistry.ensureRuleTypeEnabled(attributes.alertTypeId);
 
-    const taskDoc = attributes.scheduledTaskId
-      ? await this.taskManager.get(attributes.scheduledTaskId)
-      : null;
+    let taskDoc: ConcreteTaskInstance | null = null;
+    try {
+      taskDoc = attributes.scheduledTaskId
+        ? await this.taskManager.get(attributes.scheduledTaskId)
+        : null;
+    } catch (err) {
+      return i18n.translate('xpack.alerting.rulesClient.runSoon.getTaskError', {
```

Comment:
do we really need to return an error at this point? I am wondering if we can juts ignore `attributes.scheduledTaskId` and just use id

Reply:
If there is an error getting the task doc at this point, it likely means the task doc doesn't exist, so trying to run it later will result in an error as well. Returning at this point avoids us checking for the task again, which happens when you call `taskManager.runSoon`